x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedJob.java

```diff
@@ -151,9 +158,29 @@ long runRealtime() throws Exception {
         request.setCalcInterim(true);
         request.setAdvanceTime(String.valueOf(end));
         run(start, end, request);
+        checkForMissingDataIfNecessary();
         return nextRealtimeTimestamp();
     }
 
+    private void checkForMissingDataIfNecessary() {
+        if (isRunning() && !isIsolated && checkForMissingDataTriggered()) {
+
+            // Keep track of the last bucket time for which we did a missing data check
+            this.lastBucketDataCheckTimeMs = this.latestFinalBucketEndTimeMs;
+            List<DelayedDataDetector.BucketWithMissingData> response = delayedDataDetector.detectMissingData(latestFinalBucketEndTimeMs);
+            if (response.isEmpty() == false) {
+                long totalRecordsMissing = response.stream()
+                    .mapToLong(DelayedDataDetector.BucketWithMissingData::getMissingDocumentCount)
+                    .sum();
+                auditor.warning(jobId, Messages.getMessage(Messages.JOB_AUDIT_DATAFEED_MISSING_DATA, totalRecordsMissing));
+            }
+        }
+    }
+
+    private boolean checkForMissingDataTriggered() {
+        return this.latestFinalBucketEndTimeMs > this.lastBucketDataCheckTimeMs * BUCKETS_BETWEEN_MISSING_DATA_CHECK;
```

Comment:
Should this be 
`this.latestFinalBucketEndTimeMs > this.lastBucketDataCheckTimeMs + (bucketSpan * BUCKETS_BETWEEN_MISSING_DATA_CHECK);

Reply:
lulz, you are right, bad at math :D